src/rnp/rnp.cpp

```diff
@@ -270,37 +270,37 @@ rnp_on_signatures(pgp_parse_handler_t *handler, pgp_signature_info_t *sigs, int
         expiry = signature_get_expiration(sigs[i].sig);
 
         if (create > 0) {
-            fprintf(io->res, "%s made %s", title, ctime(&create));
+            fprintf(resfp, "%s made %s", title, ctime(&create));
             if (expiry > 0) {
                 create += expiry;
-                fprintf(io->res, "Valid until %s\n", ctime(&create));
+                fprintf(resfp, "Valid until %s\n", ctime(&create));
             }
         } else {
-            fprintf(io->res, "%s\n", title);
+            fprintf(resfp, "%s\n", title);
         }
 
         signature_get_keyid(sigs[i].sig, keyid);
-        fprintf(io->res,
+        fprintf(resfp,
                 "using %s key %s\n",
                 pgp_show_pka(sigs[i].sig->palg),
                 userid_to_id(keyid, id));
 
         if (!sigs[i].no_signer) {
-            key = rnp_key_store_get_key_by_id(io, handler->ctx->rnp->pubring, keyid, NULL);
-            repgp_print_key(io, handler->ctx->rnp->pubring, key, "signature ", 0);
+            key = rnp_key_store_get_key_by_id(handler->ctx->rnp->pubring, keyid, NULL);
+            repgp_print_key(handler->ctx->rnp->pubring, key, "signature ", 0);
         }
     }
 
     if (count == 0) {
-        fprintf(io->res, "No signature(s) found - is this a signed file?\n");
+        fprintf(stderr, "No signature(s) found - is this a signed file?\n");
```

Comment:
Should these just go to `resfp` instead?

Reply:
Not sure here - resfp should get results, but this is more like error message/message to the user (i.e. to the terminal). I'll leave this one to be resolved in another PR (see comment below).

Reply:
Agreed. Merging now.